code/obj/item/neon_lining.dm

```diff
@@ -0,0 +1,197 @@
+#define STARTLINING 1
+/obj/item/neon_lining
+	name = "neon lining"
+	var/base_name = "neon lining"
+	desc = "A coil of neon lining."
+	amount = STARTLINING
+	max_stack = MAXLINING
+	stack_type = /obj/item/neon_lining
+	icon = 'icons/obj/decals/neon_lining.dmi'
+	icon_state = "item_blue"
+	item_state = "electronic"
+	throwforce = 2
+	w_class = 1.0
+	throw_speed = 2
+	throw_range = 5
+	flags = TABLEPASS|EXTRADELAY|FPRINT|CONDUCT|ONBELT
+	stamina_damage = 5
+	stamina_cost = 5
+	stamina_crit_chance = 10
+	rand_pos = 1
+	event_handler_flags = USE_GRAB_CHOKE | USE_FLUID_ENTER
+	special_grab = /obj/item/grab
+
+	var/lining_item_color = 0
+
+	New(loc, length = STARTLINING)
+		src.amount = length
+		pixel_x = rand(-2,2)
+		pixel_y = rand(-2,2)
+		..(loc)
+		BLOCK_ROPE
+
+	before_stack(atom/movable/O as obj, mob/user as mob)
+		user.visible_message("<span class='notice'>[user] begins coiling neon lining!</span>")
+
+	after_stack(atom/movable/O as obj, mob/user as mob, var/added)
+		boutput(user, "<span class='notice'>You finish coiling neon lining.</span>")
+
+	custom_suicide = 1
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] wraps neon lining around \his neck and tightens it.</b></span>")
+		user.take_oxygen_deprivation(160)
+		SPAWN_DBG(50 SECONDS)
+			if (user && !isdead(user))
+				user.suiciding = 0
+		return 1
+
+	proc/use(var/used)
+		if (src.amount < used)
+			return 0
+		else if (src.amount == used)
+			qdel(src)
+			return 1
+		else
+			amount -= used
+			return 1
+
+	proc/take(var/amt, var/newloc)
+		if (amt > amount)
+			amt = amount
+		if (amt == amount)
+			if (ismob(loc))
+				var/mob/owner = loc
+				owner.u_equip(src)
+			loc = newloc
+			return src
+		src.use(amt)
+		var/obj/item/neon_lining/C = new /obj/item/neon_lining(newloc)
+		C.amount = amt
+		return
+
+	proc/updateicon()
+		if (lining_item_color == 0)
+			set_icon_state("item_blue")
+		else if (lining_item_color == 1)
+			set_icon_state("item_pink")
+		else if (lining_item_color == 2)
+			set_icon_state("item_yellow")
+		return
+
+/obj/item/neon_lining/cut
+	New(loc, length)
+		if (length)
+			..(loc, length)
+		else
+			..(loc, rand(1,2))
+
+/obj/item/neon_lining/cut/small
+	New(loc, length)
+		..(loc, rand(1,5))
+
+/obj/item/neon_lining/shipped
+	New(loc, length)
+		..(loc, 20)
+
+/obj/item/neon_lining/attack_self(mob/user as mob)
+	if (lining_item_color > -1 && lining_item_color < 2)
+		lining_item_color++
+	else
+		lining_item_color = 0
+	boutput(user, "You change the [base_name]'s color.")
+	updateicon()
+	return
+
+/obj/item/neon_lining/examine()
+	if (amount == 1)
+		. = list("A short piece of [base_name].")
+	else
+		. = list("A coil of [base_name]. There's [amount] length[s_es(amount)] of lining in the coil.")
+
+	if (lining_item_color == 0)
+		. += "It is blue."
+	if (lining_item_color == 1)
+		. += "It is pink."
+	if (lining_item_color == 2)
+		. += "It is yellow."
```

Comment:
Unless you have a good reason (I don't think you do) you should override `get_desc` instead. Overriding `examine` is useful for stuff like paper that's supposed to do something completely different when examined (opening a browser window) but then you lose stuff like tooltips.

Reply:
Well, that part was yoinked from cable coil code, and it has `examine` rather than `get_desc`.

Reply:
Changed <get_desc> to <examine>.